ui/packages/catalog-ui-search/src/main/webapp/component/query-add/query-add.view.js

```diff
@@ -162,6 +180,49 @@ module.exports = Marionette.LayoutView.extend({
                 }.bind(this));
         }
     },
+    getQueryAsQueryTemplate: function() {
+        const formModel = this.model.get('associatedFormModel') || new SearchFormModel();
+        const formParameters = this.queryContent.currentView.serializeTemplateParameters();
+        let filterTree = cql.simplify(formParameters.filterTree || {});
+        let filterSettings = formParameters.filterSettings || {};
+        if (filterTree.filters && filterTree.filters.length === 1) {
+            filterTree = filterTree.filters[0];
+        }
+        filterSettings.sorts = filterSettings.sorts.filter(sort => sort.attribute && sort.direction)
+            .map(sort => sort.attribute + ',' + sort.direction);
+        return filterTemplate = {
```

Comment:
This combined definition of filterTemplate and return was causing a not defined error.  I fixed it by defining filterTemplate first then returning it after

Reply:
Not sure why I ended up with `return filterTemplate = {...` instead of `return {...`